spec/services/coupon_service_spec.rb

```diff
@@ -3,67 +3,94 @@
+          coupon_service.assign_coupon
+        end
 
-      it 'assigns the user a shirt coupon' do
-        expect(user.shirt_coupon).to be_a(ShirtCoupon)
-      end
+        it 'assigns the user a shirt coupon' do
+          expect(user.shirt_coupon).to be_a(ShirtCoupon)
+        end
 
-      it 'does not assign a sticker coupon' do
-        expect(user.sticker_coupon).to eq(nil)
+        it 'does not assign a sticker coupon' do
+          expect(user.sticker_coupon).to eq(nil)
+        end
       end
-    end
 
-    context 'there are only sticker coupons available' do
-      before do
-        FactoryBot.create(:sticker_coupon)
-        coupon_service.assign_coupon
-      end
+      context 'there are only sticker coupons available' do
+        before do
+          FactoryBot.create(:sticker_coupon)
+          coupon_service.assign_coupon
+        end
 
-      it 'assigns the user a sticker coupon' do
-        expect(user.sticker_coupon).to be_a(StickerCoupon)
-      end
+        it 'assigns the user a sticker coupon' do
+          expect(user.sticker_coupon).to be_a(StickerCoupon)
+        end
 
-      it 'does not assign a shirt coupon' do
-        expect(user.shirt_coupon).to eq(nil)
+        it 'does not assign a shirt coupon' do
+          expect(user.shirt_coupon).to eq(nil)
+        end
       end
-    end
 
-    context 'there are both shirt and sticker coupons available' do
-      before do
-        FactoryBot.create(:shirt_coupon)
-        FactoryBot.create(:sticker_coupon)
-        coupon_service.assign_coupon
-      end
+      context 'there are both shirt and sticker coupons available' do
+        before do
+          FactoryBot.create(:shirt_coupon)
+          FactoryBot.create(:sticker_coupon)
+          coupon_service.assign_coupon
+        end
+
+        it 'assigns the user a shirt coupon' do
+          expect(user.shirt_coupon).to be_a(ShirtCoupon)
+        end
 
-      it 'assigns the user a shirt coupon' do
-        expect(user.shirt_coupon).to be_a(ShirtCoupon)
+        it 'does not assign a sticker coupon' do
+          expect(user.sticker_coupon).to eq(nil)
+        end
       end
 
-      it 'does not assign a sticker coupon' do
-        expect(user.sticker_coupon).to eq(nil)
+      context 'there are neither shirt nor sticker coupons available' do
+        before do
+          coupon_service.assign_coupon
+        end
+
+        it 'does not assign a shirt coupon' do
+          expect(user.shirt_coupon).to eq(nil)
+        end
+
+        it 'does not assign a sticker coupon' do
+          expect(user.sticker_coupon).to eq(nil)
+        end
       end
     end
 
-    context 'there are neither shirt nor sticker coupons available' do
-      before do
-        coupon_service.assign_coupon
-      end
+    context 'with an incomplete user' do
+      let(:user) { FactoryBot.create(:user, :incompleted) }
+      let(:coupon_service) { CouponService.new(user) }
+
+      context 'there are sticker coupons available' do
```

Comment:
This needs additional contexts as above for the various combinations of coupons being available.

More ideal would be to have this behave the same way for both `completed` and `incompleted` users.

Reply:
Added pending tests to use when `gifted_shirt` is added to the state machine